samples/web3wallet/src/main/kotlin/com/walletconnect/web3/wallet/ui/common/Namespaces.kt

```diff
@@ -4,69 +4,36 @@ import com.walletconnect.web3.wallet.client.Wallet
     val map = mutableMapOf<String, List<String>>()
-    namespace.chains.map { chain ->
-        chain to namespace.methods
-    }.plus(
-        namespace.extensions?.flatMap { extension ->
-            extension.chains.map { chain ->
-                chain to extension.methods
-            }
-        } ?: emptyList()
-    ).onEach { (chain, methods) ->
-        map[chain] = (map[chain] ?: emptyList()).plus(methods)
-    }
+    namespace.chains
+        .map { chain -> chain to namespace.methods }
+        .onEach { (chain, methods) -> map[chain] = (map[chain] ?: emptyList()).plus(methods) }
 
     return map[chainId] ?: emptyList()
 }
 
 fun getAllEventsByChainId(namespace: Wallet.Model.Namespace.Proposal, chainId: String): List<String> {
     val map = mutableMapOf<String, List<String>>()
-    namespace.chains.map { chain ->
-        chain to namespace.events
-    }.plus(
-        namespace.extensions?.flatMap { extension ->
-            extension.chains.map { chain ->
-                chain to extension.events
-            }
-        } ?: emptyList()
-    ).onEach { (chain, events) ->
-        map[chain] = (map[chain] ?: emptyList()).plus(events)
-    }
+    namespace.chains
+        .map { chain -> chain to namespace.events }
+        .onEach { (chain, events) -> map[chain] = (map[chain] ?: emptyList()).plus(events) }
 
     return map[chainId] ?: emptyList()
 }
 
-
 fun getAllMethodsByChainId(namespace: Wallet.Model.Namespace.Session, accountId: String): List<String> {
     val map = mutableMapOf<String, List<String>>()
-    namespace.accounts.map { account ->
-        account to namespace.methods
-    }.plus(
-        namespace.extensions?.flatMap { extension ->
-            extension.accounts.map { account ->
-                account to extension.methods
-            }
-        } ?: emptyList()
-    ).onEach { (account, methods) ->
-        map[account] = (map[account] ?: emptyList()).plus(methods)
-    }
+    namespace.accounts
+        .map { account -> account to namespace.methods }
+        .onEach { (account, methods) -> map[account] = (map[account] ?: emptyList()).plus(methods) }
 
     return map[accountId] ?: emptyList()
 }
 
 fun getAllEventsByChainId(namespace: Wallet.Model.Namespace.Session, accountId: String): List<String> {
     val map = mutableMapOf<String, List<String>>()
-    namespace.accounts.map { account ->
-        account to namespace.events
-    }.plus(
-        namespace.extensions?.flatMap { extension ->
-            extension.accounts.map { account ->
-                account to extension.events
-            }
-        } ?: emptyList()
-    ).onEach { (account, events) ->
-        map[account] = (map[account] ?: emptyList()).plus(events)
-    }
+    namespace.accounts
+        .map { account -> account to namespace.events }
+        .onEach { (account, events) -> map[account] = (map[account] ?: emptyList()).plus(events) }
```

Comment:
this can be simplified without having to do any mapping

```suggestion
return if (namespace.accounts.any { account -> account == accountId }) {
        namespace.events
    } else {
        emptyList()
    }
```

Reply:
you could even do 

```suggestion
   return namespace.events.takeIf { namespace.accounts.contains(accountId) } ?: emptyList()
```